overlord/ifacestate/ifacemgr.go

```diff
@@ -155,6 +155,39 @@ func (m *InterfaceManager) Repository() *interfaces.Repository {
 	return m.repo
 }
 
+type ConnectionState struct {
+	// Auto indicates whether the connection was established automatically
+	Auto bool
+	// ByGadget indicates whether the connection was trigged by the gadget
+	ByGadget bool
+	// Interface name of the connection
+	Interface string
+	// Undesired indicates whether the connection was explicitly
+	// disconnected
+	Undesired bool
+}
```

Comment:
Some suggestion on the shape of this API:

```
// IsConnected returns true if a connection is established.
func (cst *ConnectionState) IsConnected() bool {
  return !cst.Undesired
}

// ConnectionInitiator returns the entity that caused a connection to be formed. 
func (cst *ConnectionState) ConnectionInitiator () string {
  if !cst.IsConnected() {
    return ""
  }
  if cst.Auto {
    return "snapd"
  }
  if cst.ByGadget {
    return "gadget"
  }
  return "user"
}
```

I think that the state we keep is fine but spreading the manipulation of those fields across the tree will end up badly as we evolve that state.

Reply:
Need to think about. I think we can come back to this once I update the daemon bits. It may happen that the snapd API will expose an aggregate string that indicates who trigged the connection (like you're suggesting now), instead of individual fields.

Reply:
On a side note, we will soon have auto-connections initiated by hotplug, but I'm not sure we will want to differentiate them. In any case, a note to myself, mostly ;)